tick/survival/convolutional_sccs.py

```diff
@@ -0,0 +1,922 @@
+        '_fitted',
+        '_step_size',
+        # refit coeffs, median, and CI data
+        'bootstrap_coeffs',
+    ]
+
+    _attrinfos = {key: {'writable': False} for key in _const_attr}
+
+    def __init__(self, n_lags: np.array,
+                 penalized_features: np.array,
+                 strength_tv=None, strength_group_l1=None,
+                 step: float = None, tol: float = 1e-5, max_iter: int = 100,
+                 verbose: bool = False, print_every: int = 10,
+                 record_every: int = 10, random_state: int = None):
+        """ConvSCCS model. This class allows to estimate a lagged effect for
+        each feature. TV and Group L1 penalties might be used to penalize the
+        coefficient groups modelling the lagged effects.
+
+        Parameters
+        ----------
+        n_lags : `numpy.ndarray`, shape=(n_features,), dtype="uint64"
+            Number of lags per feature. The model will regress labels on the
+            last observed values of the features over their corresponding
+            `n_lags` time intervals. `n_lags` values must be between 0 and
+            `n_intervals` - 1.
+
+        penalized_features : `numpy.ndarray`, shape=(n_features,), dtype="bool"
+            Booleans indicating whether the features should be penalised or
+            not.
+
+        strength_tv : `float`, default=None
+            Strength of the TV penalization. This value should be `None` or
+            greater than 0.
+
+        strength_group_l1 : `float`, default=None
+            Strength of the group Lasso penalization. This value should be
+            `None` or greater than 0.
+
+        step : `float`
+            Step-size parameter, the most important parameter of the solver.
+            Whenever possible, this can be automatically tuned as
+            ``step = 1 / model.get_lip_max()``. Otherwise, use a
+            try-an-improve approach.
+
+        tol : `float`, default=1e-5
+            The tolerance of the solver (iterations stop when the stopping
+            criterion is below it).
+
+        max_iter : `int`, default=100
+            Maximum number of iterations of the solver, namely maximum
+            number of epochs.
+
+        verbose : `bool`, default=False
+            If `True`, solver verboses history, otherwise nothing is
+            displayed.
+
+        random_state : `int`, default=None
+            If not None, the seed of the random sampling.
+
+
+        Attributes
+        ----------
+        n_cases : `int` (read-only)
+            Number of samples
+
+        n_intervals : `int` (read-only)
+            Number of time intervals
+
+        n_features : `int` (read-only)
+            Number of features
+
+        n_coeffs : `int` (read-only)
+            Total number of coefficients of the model
+
+        coeffs : `numpy.ndarray`, shape=(n_coeffs,), dtype="float64" (read-only)
+            Coefficients of the model.
+
+        bootstrap_coeffs : `Bootstrap_CI` (read-only)
```

Comment:
We need to explain a little bit about Bootstrap_CI... We should rename it confidence_intervals

Reply:
And this should really be called "confidence_intervals"